File: Alcoolougasolina/app/src/main/java/com/example/alcoolougasolina/MainActivity.java
package com.example.alcoolougasolina;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {

    private TextInputEditText editPrecoAlcool, editPrecoGasolina;
    private TextView textResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editPrecoAlcool = findViewById(R.id.idPrecoAlcool);
        editPrecoGasolina = findViewById(R.id.idPrecoGasolina);

        textResultado = findViewById(R.id.textResultado);

    }

    public void calcularPreco(View view){

        //recuperar dados digitados
        String precoAlcool = editPrecoAlcool.getText().toString();
        String precoGasolina = editPrecoGasolina.getText().toString();

        //validar campos digitados
        Boolean camposValidados = validarCampos(precoAlcool, precoGasolina);

        if ( camposValidados ) {

            // convertendo string para numeros
            Double valorAlcool = Double.parseDouble(precoAlcool);
            Double valorGasolina = Double.parseDouble(precoGasolina);

            /* calculo (valorAlcool / valorGasolina) >= 0.7 melhor gasolina
            senao melhor alcool  */

            Double resultado = (valorAlcool/valorGasolina);
            if ( resultado >= 0.7 ){
                textResultado.setText("Melhor utilizar gasolina.");
            }else {
                textResultado.setText("Melhor utilizar alcool.");
            }

        }else {
            textResultado.setText("Preencha os campos primeiro!");
        }

    }

    public Boolean validarCampos(String pAlcool, String pGasolina){

        Boolean camposValidados = true;


        if( pAlcool == null || pAlcool.equals("")){
            camposValidados = false;

        }else if (pGasolina == null || pGasolina.equals("")){
            camposValidados = false;

        }

        return camposValidados;

    }
}

File: AlcoolOuGasolinaKotlin/app/src/main/java/com/example/alcoolougasolinakotlin/MainActivity.kt
package com.example.alcoolougasolinakotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import com.example.alcoolougasolinakotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.buttonCalcular.setOnClickListener { calcularPreco() }
    }

    private fun calcularPreco() {
        var precoAlcool = findViewById<EditText>(R.id.editTextAlcool).toString()
        var precoGasolina = findViewById<EditText>(R.id.editTextGasolina).toString()

        val validaCampos = validarCampos(precoAlcool, precoGasolina)

        if ( validaCampos ) {
            calcularMelhorPreco( precoAlcool, precoGasolina)
        } else binding.textViewResultado.text = "Preencha os dados primeiro!"

    }

    private fun validarCampos(precoAlcool: String, precoGasolina: String): Boolean {

        var camposValidados: Boolean = true

        if ( precoAlcool== null || precoAlcool.equals("") ) {
            camposValidados = false
        } else if (precoGasolina== null || precoGasolina.equals("")){
            camposValidados = false
        }

        return camposValidados
    }

    private fun calcularMelhorPreco(precoAlcool: String, precoGasolina: String) {

        val valorAlcool = precoAlcool.toDouble()
        val valorGasolina = precoGasolina.toDouble()

        /*
            (precoAlcool / precoGasolina) se resultado for
            maior 0.7 melhor gasolina se nao alcool
         */

        var resultado = valorAlcool / valorGasolina

        if ( resultado >= 0.7) {
            binding.textViewResultado.text = "Melhor usar Alcool!"
        } else {
            binding.textViewResultado.text = "melhor usar Gasolina!"
        }
    }


}